### PR TITLE
Fix tests and rely on `_match_id` for some extractors

### PR DESCRIPTION
**Pull request by TRox1972**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR contains multiple changes for some extractors that seemed too small for one PR each. The changes are
- Use `_match_id` when `id` is the only captured group in `_VALID_URL`
- Use `_parse_json` instead of `json.loads`
- Use `# coding: utf-8`
- Fixed some tests
- Improved some regexes

I think these changes are simple and obvious improvements, and can therefore be accepted as one PR. But if some changes aren't, let me know and I'll remove them.

### CODE REVIEW

File: youtube_dl/extractor/reverbnation.py [hunk 1]
```diff
@@ -10,20 +8,18 @@ class ReverbNationIE(InfoExtractor):
...
         'url': 'http://www.reverbnation.com/alkilados/song/16965047-mona-lisa',
-        'md5': '3da12ebca28c67c111a7f8b262d3f7a7',
+        'md5': 'c0aaf339bcee189495fdf5a8c8ba8645',
         'info_dict': {
             'id': '16965047',
             'ext': 'mp3',
             'title': 'MONA LISA',
             'uploader': 'ALKILADOS',
             'uploader_id': '216429',
-            'thumbnail': 're:^https://gp1\.wac\.edgecastcdn\.net/.*?\.jpg$'
```
Comment (yan12125) [inline youtube_dl/extractor/reverbnation.py]: It's better to fix thumbnail extraction instead of remove the test.

Comment (TRox1972) [inline youtube_dl/extractor/reverbnation.py]: Should I create a new PR for that?

Comment (yan12125) [inline youtube_dl/extractor/reverbnation.py]: Put it in this one is OK.

Comment (TRox1972) [inline youtube_dl/extractor/reverbnation.py]: done.

File: youtube_dl/extractor/reverbnation.py [hunk 2]
```diff
@@ -10,35 +8,40 @@ class ReverbNationIE(InfoExtractor):
...
             song_id,
             note='Downloading information of song %s' % song_id
         )
 
+        thumbnails = [{
+            'url': api_res.get('image'),
+        }, {
+            'url': api_res.get('thumbnail'),
+            'preference': -2,
+        }]
```
Comment (dstftw) [inline youtube_dl/extractor/reverbnation.py]: This change is not related to PR's purpose. Don't mix unrelated changes in single PR.

Comment (TRox1972) [inline youtube_dl/extractor/reverbnation.py]: Should I move the thumbnail fix to another PR?

Comment (yan12125) [inline youtube_dl/extractor/reverbnation.py]: My concern was there are too many small PRs. Moving them out of this one is of course cleaner.

Comment (dstftw) [inline youtube_dl/extractor/reverbnation.py]: Each PR should be granular enough and solve some particular problem. For example, contain either only homogeneous changes (e.g. `_match_id` refactoring) across whole code base or heterogeneous changes for a single extractor.

Comment (TRox1972) [inline youtube_dl/extractor/reverbnation.py]: OK. I'll close this and do it individually for extractors with more than the `_match_id` change
